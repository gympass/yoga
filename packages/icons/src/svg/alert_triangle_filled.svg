<svg width="12" height="12" viewBox="0 0 12 12" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M0.15618 10.0075C-0.301039 10.9036 0.300392 12 1.24914 12H10.7509C11.6996 12 12.301 10.9036 11.8438 10.0075L7.09296 0.696582C6.61905 -0.232195 5.38095 -0.232194 4.90704 0.696583L0.15618 10.0075ZM6.02368 7.23909C6.36813 7.23909 6.64736 6.9381 6.64736 6.56682V5.22227C6.64736 4.85099 6.36813 4.55 6.02368 4.55C5.67923 4.55 5.4 4.85099 5.4 5.22227V6.56682C5.4 6.9381 5.67923 7.23909 6.02368 7.23909ZM6.02368 8.58363C5.67923 8.58363 5.4 8.88462 5.4 9.2559C5.4 9.62719 5.67923 9.92817 6.02368 9.92817C6.36813 9.92817 6.64736 9.62719 6.64736 9.2559C6.64736 8.88462 6.36813 8.58363 6.02368 8.58363Z" />
</svg>
