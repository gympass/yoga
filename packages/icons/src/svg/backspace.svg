<svg width="25" height="18" viewBox="0 0 25 18" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M22.166 0H7.26602C6.56602 0 5.96602 0.3 5.56602 0.9L0.166016 9L5.56602 17.1C5.96602 17.6 6.56602 18 7.26602 18H22.166C23.266 18 24.166 17.1 24.166 16V2C24.166 0.9 23.266 0 22.166 0ZM19.166 12.6L17.766 14L14.166 10.4L10.566 14L9.16602 12.6L12.766 9L9.16602 5.4L10.566 4L14.166 7.6L17.766 4L19.166 5.4L15.566 9L19.166 12.6Z" fill="#57544D"/>
</svg>
