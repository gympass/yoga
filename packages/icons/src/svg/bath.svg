<svg xmlns="http://www.w3.org/2000/svg" height="24" width="24" viewBox="0 0 24 24"><path d="M6.083.443C4.837.688 3.812 1.671 3.477 2.94c-.12.456-.121 1.116-.002 1.6.243.99.909 1.808 1.825 2.243.561.266.788.315 1.46.316.543 0 .606-.009.98-.128.556-.178.967-.43 1.384-.847.417-.417.669-.828.847-1.384.119-.374.128-.437.128-.98-.001-.664-.049-.893-.304-1.44A3.357 3.357 0 0 0 7.487.464C7.18.395 6.392.383 6.083.443M7.413 2c.504.191.967.677 1.126 1.181.054.172.066.321.054.651-.013.376-.03.459-.137.689-.358.76-1.192 1.214-1.99 1.082-1.068-.176-1.749-1.167-1.545-2.248.122-.651.754-1.298 1.401-1.434.288-.06.827-.022 1.091.079M5.6 7.161c-.673.117-1.297.591-1.616 1.228-.207.414-.224.562-.224 1.959v1.292l-.91.001c-1.041 0-1.276.038-1.689.268a2.634 2.634 0 0 0-.874.857c-.245.431-.287.714-.287 1.91 0 1.217.044 1.585.263 2.242.298.889.8 1.665 1.597 2.464.669.672 1.564 1.32 2.353 1.703l.188.091-.677.682c-.647.652-.678.691-.706.876-.057.388.179.768.537.865.389.104.557-.005 1.56-1.012.804-.807.817-.818.94-.782.294.087.876.204 1.282.257.645.085 8.681.085 9.326 0 .406-.053.988-.17 1.282-.257.123-.036.136-.025.94.782 1.003 1.007 1.171 1.116 1.56 1.012.358-.097.594-.477.537-.865-.028-.185-.059-.224-.706-.876l-.677-.682.191-.092c.78-.378 1.697-1.044 2.37-1.721.778-.782 1.282-1.565 1.577-2.445.219-.657.263-1.025.263-2.242 0-1.196-.042-1.479-.287-1.91a2.634 2.634 0 0 0-.874-.857c-.481-.268-.475-.268-3.449-.268-1.468-.001-2.67-.011-2.67-.024a.71.71 0 0 1 .096-.15c.393-.524.534-1.339.34-1.973a2.578 2.578 0 0 0-.66-1.047c-.661-.589-1.694-.733-2.463-.344-.113.058-.828.52-1.589 1.029-1.461.975-1.601 1.047-1.901.98-.2-.045-.17-.022-1.743-1.331a151.145 151.145 0 0 0-1.467-1.212c-.447-.35-1.137-.512-1.733-.408m.691 1.533c.071.03.798.613 1.617 1.294 1.843 1.536 2.009 1.632 2.812 1.632.653 0 .762-.052 2.407-1.146a67.806 67.806 0 0 1 1.549-1.012.87.87 0 0 1 .332-.061.73.73 0 0 1 .562 1.203c-.093.111-2.575 1.783-3.09 2.082-.244.142-.798.333-1.145.394-.444.079-1.231.034-1.667-.096-.617-.184-.979-.395-1.688-.986-.404-.337-.446-.358-.719-.358-.251 0-.39.058-.546.226-.156.169-.195.267-.195.493 0 .309.101.461.566.847.939.778 1.587 1.112 2.565 1.32.54.115 1.565.124 2.049.018.81-.177 1.36-.425 2.285-1.03l.604-.396 3.699.011 3.698.011.166.11c.312.207.328.278.328 1.483 0 .981-.007 1.084-.095 1.432-.123.483-.279.825-.599 1.307-.99 1.497-2.984 2.706-5.066 3.074-.391.069-.712.074-4.72.074s-4.329-.005-4.72-.074c-2.521-.445-4.841-2.104-5.54-3.962-.192-.51-.22-.749-.22-1.868 0-1.186.016-1.26.328-1.466l.166-.11 1.363-.02c1.448-.021 1.494-.028 1.668-.226.194-.22.195-.232.196-1.922 0-.876.017-1.66.036-1.744a.75.75 0 0 1 1.014-.534" fill-rule="evenodd"/></svg>