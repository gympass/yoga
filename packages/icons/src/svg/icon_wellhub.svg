<svg width="32" height="32" viewBox="0 0 32 32" fill="white" xmlns="http://www.w3.org/2000/svg">
  <path d="M17.7034 15.1737L17.6974 15.1559C17.6312 14.9871 17.6613 14.7976 17.7725 14.6525C20.7708 10.7972 17.1139 3.08051 14.7683 0.853755C14.0495 0.1727 13.0752 -0.120463 12.0918 0.04536C11.1054 0.214143 10.2844 0.815252 9.83632 1.6947L9.59273 2.18329L9.64085 2.16553C9.23487 3.424 9.77918 4.83052 11.0783 5.78103C11.3069 5.94982 11.5715 6.13637 11.8602 6.34069C13.7247 7.65838 16.8493 9.8644 17.1711 11.8868C17.2613 12.4583 17.129 12.9706 16.7651 13.4592C16.6208 13.6517 16.3621 13.7316 16.1366 13.6546L16.1185 13.6487C15.9261 13.5836 15.7847 13.4414 15.7396 13.2727C14.4736 8.61483 6.09827 6.64271 2.90153 7.05727C1.91213 7.1846 1.07009 7.75313 0.591933 8.61186C0.110768 9.4765 0.0776879 10.4803 0.501715 11.3716L0.531788 11.4338C1.17835 12.8048 2.78424 13.4207 4.52847 12.9706C4.80514 12.8996 5.11789 12.8107 5.45772 12.716C8.36877 11.9076 13.2406 10.5573 14.1397 13.243C14.227 13.5066 14.1247 13.7908 13.8992 13.9419L13.8721 13.9596C13.7097 14.0692 13.4992 14.087 13.3157 14.0129C8.82287 12.183 2.08655 17.359 0.447584 20.0329C-0.0666613 20.8709 -0.141843 21.8688 0.243089 22.7749C0.628021 23.6869 1.40991 24.3413 2.38427 24.5723L2.45044 24.5871C3.94505 24.9424 5.43667 24.0925 6.16142 22.4728C6.27871 22.2152 6.40501 21.9161 6.54335 21.5934C7.72521 18.8307 9.704 14.2113 12.4286 15.2773C12.6632 15.3691 12.8105 15.609 12.7955 15.8784C12.7835 16.0828 12.6692 16.2663 12.4948 16.3581C8.20938 18.65 8.12518 27.0714 9.23487 30.0148C9.5807 30.9327 10.3295 31.6108 11.2888 31.8803C12.2542 32.1497 13.2586 31.9602 14.0555 31.3532L14.1067 31.3147C15.3216 30.3879 15.5772 28.7119 14.7412 27.1425C14.6089 26.8938 14.4525 26.6125 14.2811 26.3045C12.8135 23.6573 10.3566 19.2245 12.9669 17.8209C13.1744 17.7084 13.439 17.741 13.6225 17.9038L13.6435 17.9216C13.7939 18.0608 13.8631 18.2621 13.827 18.4605C12.9699 23.1776 19.616 28.5046 22.6534 29.4877C23.6007 29.7956 24.6111 29.6417 25.4231 29.0702C26.238 28.4987 26.7102 27.6044 26.7252 26.6213L26.7252 26.5592C26.7252 26.5592 26.7252 26.5296 26.7252 26.5177C26.7252 25.0253 25.5584 23.8024 23.8052 23.4707C23.5225 23.4174 23.2007 23.3612 22.8488 23.299C20.5573 22.9022 16.72 22.2389 15.5802 20.5185C15.2494 20.0181 15.1622 19.4673 15.3216 18.8336C15.3667 18.653 15.5291 18.5109 15.7216 18.4842L15.7727 18.4753C15.9471 18.4487 16.1246 18.5168 16.2448 18.6589C19.4025 22.3899 27.871 20.6044 30.5986 18.8484C31.4316 18.3125 31.9399 17.4449 31.994 16.4647C31.997 16.4085 32 16.3522 32 16.293C32 15.3721 31.591 14.5193 30.8482 13.9241L30.8061 13.8915C29.6182 12.9321 27.8981 13.0595 26.5298 14.2113C26.3132 14.3949 26.0696 14.6052 25.802 14.8361C24.0578 16.3522 21.1347 18.8869 19.0536 18.6915C18.4462 18.6352 17.956 18.3569 17.55 17.8387C17.4267 17.6818 17.4147 17.4567 17.5199 17.279C17.6132 17.1221 17.7876 17.0244 17.9831 17.0185C22.927 16.9267 26.7854 9.29885 27.0921 6.10677C27.1853 5.13256 26.8124 4.1998 26.0696 3.54836C25.3208 2.89099 24.3284 2.63337 23.348 2.84065L23.2789 2.85545C21.7752 3.17229 20.8099 4.56106 20.864 6.34069C20.8731 6.62199 20.8911 6.9418 20.9091 7.29121C21.0234 9.57127 21.2189 13.3881 19.7724 14.8687C19.3514 15.301 18.8281 15.5053 18.1695 15.4994C17.962 15.4994 17.7786 15.3662 17.7064 15.1678L17.7034 15.1737Z" />
</svg>
