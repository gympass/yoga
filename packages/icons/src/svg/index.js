import Accessibility from './accessibility.svg';
import Accessible from './accessible.svg';
import AddBooking from './add_booking.svg';
import Add from './add.svg';
import AlertCircle from './alert_circle.svg';
import AlertTriangle from './alert_triangle.svg';
import AlertTriangleFilled from './alert_triangle_filled.svg';
import ArrowDropDown from './arrow_drop_down.svg';
import ArrowDropUp from './arrow_drop_up.svg';
import ArrowLeft from './arrow_left.svg';
import ArrowRight from './arrow_right.svg';
import BankStatement from './bank_statement.svg';
import BarChart from './bar_chart.svg';
import BookingFilled from './booking_filled.svg';
import Booking from './booking.svg';
import BuildingFilled from './building_filled.svg';
import Building from './building.svg';
import BulletList from './bullet_list.svg';
import Camera from './camera.svg';
import Cart from './cart.svg';
import Chat from './chat.svg';
import CheckFilled from './check_filled.svg';
import Check from './check.svg';
import CheckedFull from './checked_full.svg';
import CheckedHalf from './checked_half.svg';
import CheckinFilled from './checkin_filled.svg';
import Checkin from './checkin.svg';
import ChevronDown from './chevron_down.svg';
import ChevronLeft from './chevron_left.svg';
import ChevronRight from './chevron_right.svg';
import ChevronUp from './chevron_up.svg';
import CloseFilled from './close_filled.svg';
import Close from './close.svg';
import CompassFilled from './compass_filled.svg';
import Compass from './compass.svg';
import Copy from './copy.svg';
import CornerUpLeft from './corner_up_left.svg';
import CornerUpRight from './corner_up_right.svg';
import CreditCard from './credit_card.svg';
import Disconnect from './disconnect.svg';
import DislikeFilled from './dislike_filled.svg';
import Dislike from './dislike.svg';
import Doc from './doc.svg';
import Download from './download.svg';
import Draggable from './draggable.svg';
import Dumbbell from './dumbbell.svg';
import DumbbellFilled from './dumbbell_filled.svg';
import Edit from './edit.svg';
import Exit from './exit.svg';
import Facebook from './facebook.svg';
import FamilyMember from './family_member.svg';
import FavoriteFilled from './favorite_filled.svg';
import Favorite from './favorite.svg';
import Filter from './filter.svg';
import Google from './google.svg';
import Headphone from './headphone.svg';
import Heartbeat from './heartbeat.svg';
import HelpCircle from './help_circle.svg';
import Help from './help.svg';
import History from './history.svg';
import HomeFilled from './home_filled.svg';
import Home from './home.svg';
import Image from './image.svg';
import Info from './info.svg';
import Instagram from './instagram.svg';
import JumpRope from './jump_rope.svg';
import LikeFilled from './like_filled.svg';
import Like from './like.svg';
import Link from './link.svg';
import Linkedin from './linkedin.svg';
import Location from './location.svg';
import LockOff from './lock_off.svg';
import Lock from './lock.svg';
import LockerRoom from './locker_room.svg';
import LogOut from './log_out.svg';
import Loop from './loop.svg';
import MailRead from './mail_read.svg';
import MailUnread from './mail_unread.svg';
import MapPin from './map_pin.svg';
import Map from './map.svg';
import MenuList from './menu_list.svg';
import MenuMore from './menu_more.svg';
import Money from './money.svg';
import Monitor from './monitor.svg';
import NetworkOff from './network_off.svg';
import NetworkOn from './network_on.svg';
import NotificationFilled from './notification_filled.svg';
import NotificationOff from './notification_off.svg';
import PaperPin from './paper_pin.svg';
import Paperclip from './paperclip.svg';
import Password from './password.svg';
import Pause from './pause.svg';
import PhoneFilled from './phone_filled.svg';
import Phone from './phone.svg';
import PieChart from './pie_chart.svg';
import PinFilled from './pin_filled.svg';
import Play from './play.svg';
import PlayFilled from './play_filled.svg';
import PlusCircle from './plus_circle.svg';
import Printer from './printer.svg';
import QRCodeReader from './qr_code_reader.svg';
import QRCode from './qr_code.svg';
import Refresh from './refresh.svg';
import SearchFilled from './search_filled.svg';
import Search from './search.svg';
import Settings from './settings.svg';
import ShareAndroid from './share_android.svg';
import ShareIos from './share_ios.svg';
import ShieldCheck from './shield_check.svg';
import ShieldSecure from './shield_secure.svg';
import Signal from './signal.svg';
import SmartphoneFilled from './smartphone_filled.svg';
import Smartphone from './smartphone.svg';
import SortAscending from './sort_ascending.svg';
import SortDescending from './sort_descending.svg';
import StarFilled from './star_filled.svg';
import Star from './star.svg';
import PriceTag from './price_tag.svg';
import Time from './time.svg';
import Token from './token.svg';
import Trash from './trash.svg';
import TrendingUp from './trending_up.svg';
import TriangleAlert from './triangle_alert.svg';
import Twitter from './twitter.svg';
import Upload from './upload.svg';
import Upload2 from './upload2.svg';
import UserAdd from './user_add.svg';
import UserFilled from './user_filled.svg';
import User from './user.svg';
import Video from './video.svg';
import VisibilityOff from './visibility_off.svg';
import Visibility from './visibility.svg';
import Wallet from './wallet.svg';
import Whatsapp from './whatsapp.svg';
import Youtube from './youtube.svg';

export {
  Accessibility,
  Accessible,
  AddBooking,
  Add,
  AlertCircle,
  AlertTriangle,
  AlertTriangleFilled,
  ArrowDropDown,
  ArrowDropUp,
  ArrowLeft,
  ArrowRight,
  BankStatement,
  BarChart,
  BookingFilled,
  Booking,
  BuildingFilled,
  Building,
  BulletList,
  Camera,
  Cart,
  Chat,
  CheckFilled,
  Check,
  CheckedFull,
  CheckedHalf,
  CheckinFilled,
  Checkin,
  ChevronDown,
  ChevronLeft,
  ChevronRight,
  ChevronUp,
  CloseFilled,
  Close,
  CompassFilled,
  Compass,
  Copy,
  CornerUpLeft,
  CornerUpRight,
  CreditCard,
  Disconnect,
  DislikeFilled,
  Dislike,
  Doc,
  Download,
  Draggable,
  Dumbbell,
  DumbbellFilled,
  Edit,
  Exit,
  Facebook,
  FamilyMember,
  FavoriteFilled,
  Favorite,
  Filter,
  Google,
  Headphone,
  Heartbeat,
  HelpCircle,
  Help,
  History,
  HomeFilled,
  Home,
  Image,
  Info,
  Instagram,
  JumpRope,
  LikeFilled,
  Like,
  Link,
  Linkedin,
  Location,
  LockOff,
  Lock,
  LockerRoom,
  LogOut,
  Loop,
  MailRead,
  MailUnread,
  MapPin,
  Map,
  MenuList,
  MenuMore,
  Money,
  Monitor,
  NetworkOff,
  NetworkOn,
  NotificationFilled,
  NotificationOff,
  PaperPin,
  Paperclip,
  Password,
  Pause,
  PhoneFilled,
  Phone,
  PieChart,
  PinFilled,
  Play,
  PlayFilled,
  PlusCircle,
  Printer,
  QRCodeReader,
  QRCode,
  Refresh,
  SearchFilled,
  Search,
  Settings,
  ShareAndroid,
  ShareIos,
  ShieldCheck,
  ShieldSecure,
  Signal,
  SmartphoneFilled,
  Smartphone,
  SortAscending,
  SortDescending,
  StarFilled,
  Star,
  PriceTag,
  Time,
  Token,
  Trash,
  TrendingUp,
  TriangleAlert,
  Twitter,
  Upload,
  Upload2,
  UserAdd,
  UserFilled,
  User,
  Video,
  VisibilityOff,
  Visibility,
  Wallet,
  Whatsapp,
  Youtube,
};
