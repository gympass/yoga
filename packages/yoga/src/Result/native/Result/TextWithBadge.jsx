import React, { useCallback, useState } from 'react';
import { useWindowDimensions } from 'react-native';
import { string, number, node } from 'prop-types';

import Text from '../../../Text';
import Box from '../../../Box';
import Badge from '../Badge';

const SCREEN_PADDINGS = 20;
const CONTENT_MARGINS = 20;
const AVATAR_CONTENT_MARGINS = 16;
const BADGE_LIMIT = 20;

const TextWithBadge = ({ avatarWidth, badgeIcon, title }) => {
  const [textSize, setTextSize] = useState(0);
  const { width: windowWidth } = useWindowDimensions();

  const textMaxSize =
    windowWidth -
    (SCREEN_PADDINGS + CONTENT_MARGINS + AVATAR_CONTENT_MARGINS + avatarWidth);
  const shouldTruncate = textSize >= textMaxSize - BADGE_LIMIT;
  const containerWidth = shouldTruncate ? null : textSize;
  const textWidth = shouldTruncate ? '100%' : null;

  const onTextLayout = useCallback(event => {
    const { width } = event.nativeEvent.layout;

    setTextSize(width);
  }, []);

  return (
    <Box
      flexDirection="row"
      alignItems="center"
      justifyContent="flex-end"
      position="relative"
      width={containerWidth}
    >
      <Text.Body1
        onLayout={onTextLayout}
        bold
        position="absolute"
        left={0}
        numberOfLines={1}
        paddingRight="medium"
        flex={1}
        width={textWidth}
      >
        {title}
      </Text.Body1>
      <Badge
        icon={badgeIcon}
        fill="text.primary"
        ml="xxxsmall"
        bg="neon"
        justifyContent="center"
        alignItems="center"
        borderRadius="circle"
        w="small"
        h="small"
      />
    </Box>
  );
};

TextWithBadge.propTypes = {
  avatarWidth: number.isRequired,
  badgeIcon: node.isRequired,
  title: string.isRequired,
};

TextWithBadge.defaultProps = {};

export default TextWithBadge;
