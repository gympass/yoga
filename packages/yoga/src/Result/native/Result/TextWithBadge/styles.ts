import React from 'react';
import styled from 'styled-components';
import Text from '../../../../Text';
import Box from '../../../../Box';

export const StyledBoxContainer = styled(Box)`
  flex-direction: row;
  align-items: center;
`;

export const StyledText = styled(Text.Body1)<{
  numberOfLines: number;
  bold: boolean;
  children: React.ReactNode;
}>`
  flex-shrink: 1;
`;
