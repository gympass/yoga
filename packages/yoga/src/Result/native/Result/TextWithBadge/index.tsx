import React, { ReactNode, useCallback, useState } from 'react';
import { useWindowDimensions } from 'react-native';

import Badge from '../../Badge';

import { StyledBoxContainer, StyledText } from './styles';

interface TextWithBadgeProps {
  avatarWidth: number;
  badgeIcon: ReactNode;
  title: string;
}

const SCREEN_PADDINGS = 20;
const CONTENT_MARGINS = 20;
const AVATAR_CONTENT_MARGINS = 16;
const BADGE_LIMIT = 20;

const TextWithBadge = ({
  avatarWidth,
  badgeIcon,
  title,
}: TextWithBadgeProps) => {
  const [textSize, setTextSize] = useState<number>(0);
  const { width: windowWidth } = useWindowDimensions();

  const textMaxSize =
    windowWidth -
    (SCREEN_PADDINGS + CONTENT_MARGINS + AVATAR_CONTENT_MARGINS + avatarWidth);
  const shouldTruncate = textSize >= textMaxSize - BADGE_LIMIT;
  const containerWidth = shouldTruncate ? null : textSize;
  const textWidth = shouldTruncate ? '100%' : null;

  const onTextLayout = useCallback(
    ({
      nativeEvent: {
        layout: { width },
      },
    }) => {
      setTextSize(width);
    },
    [],
  );

  return (
    <StyledBoxContainer containerWidth={containerWidth}>
      <StyledText
        onLayout={onTextLayout}
        bold
        numberOfLines={1}
        textWidth={textWidth}
      >
        {title}
      </StyledText>
      <Badge
        icon={badgeIcon}
        fill="text.primary"
        ml="xxxsmall"
        bg="neon"
        justifyContent="center"
        alignItems="center"
        borderRadius="circle"
        w="small"
        h="small"
      />
    </StyledBoxContainer>
  );
};

export default TextWithBadge;
