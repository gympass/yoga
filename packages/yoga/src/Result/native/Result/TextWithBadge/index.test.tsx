import React from 'react';
import { render } from '@testing-library/react-native';
import { WellhubIcon } from '@gympass/yoga-icons/src/svg/icon_wellhub.svg';

import { ThemeProvider } from '../../../../index';
import TextWithBadge from '.';

describe('TextWithBadge', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <TextWithBadge
          avatarWidth={50}
          badgeIcon={WellhubIcon}
          title="Text with badge"
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot with long title', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <TextWithBadge
          avatarWidth={50}
          badgeIcon={WellhubIcon}
          title="This is an example of a very long title that should be truncated"
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should match snapshot without badgeIcon', () => {
    const { toJSON } = render(
      <ThemeProvider>
        <TextWithBadge
          avatarWidth={50}
          badgeIcon={null}
          title="Title without Badge"
        />
      </ThemeProvider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
