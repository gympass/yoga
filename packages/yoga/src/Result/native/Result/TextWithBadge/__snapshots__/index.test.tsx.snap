// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithBadge should match snapshot 1`] = `
<View
  containerWidth={0}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "position": "relative",
        "width": 0,
      },
    ]
  }
>
  <Text
    bold={true}
    numberOfLines={1}
    onLayout={[Function]}
    style={
      [
        {
          "color": "#231B22",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "500",
          "left": 0,
          "lineHeight": 24,
          "paddingRight": 20,
          "position": "absolute",
          "width": "",
        },
      ]
    }
    textWidth={null}
  >
    Text with badge
  </Text>
  <View
    alignItems="center"
    bg="neon"
    borderRadius="circle"
    h="small"
    justifyContent="center"
    ml="xxxsmall"
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#DCFF79",
          "borderRadius": 9999,
          "height": 16,
          "justifyContent": "center",
          "marginLeft": 4,
          "width": 16,
        },
      ]
    }
    w="small"
  >
    <View
      aria-hidden={true}
      fill="#231B22"
      height={10.67}
      style={
        [
          {
            "height": 10.67,
            "width": 10.67,
          },
        ]
      }
      width={10.67}
    />
  </View>
</View>
`;

exports[`TextWithBadge should match snapshot with long title 1`] = `
<View
  containerWidth={0}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "position": "relative",
        "width": 0,
      },
    ]
  }
>
  <Text
    bold={true}
    numberOfLines={1}
    onLayout={[Function]}
    style={
      [
        {
          "color": "#231B22",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "500",
          "left": 0,
          "lineHeight": 24,
          "paddingRight": 20,
          "position": "absolute",
          "width": "",
        },
      ]
    }
    textWidth={null}
  >
    This is an example of a very long title that should be truncated
  </Text>
  <View
    alignItems="center"
    bg="neon"
    borderRadius="circle"
    h="small"
    justifyContent="center"
    ml="xxxsmall"
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#DCFF79",
          "borderRadius": 9999,
          "height": 16,
          "justifyContent": "center",
          "marginLeft": 4,
          "width": 16,
        },
      ]
    }
    w="small"
  >
    <View
      aria-hidden={true}
      fill="#231B22"
      height={10.67}
      style={
        [
          {
            "height": 10.67,
            "width": 10.67,
          },
        ]
      }
      width={10.67}
    />
  </View>
</View>
`;

exports[`TextWithBadge should match snapshot without badgeIcon 1`] = `
<View
  containerWidth={0}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "position": "relative",
        "width": 0,
      },
    ]
  }
>
  <Text
    bold={true}
    numberOfLines={1}
    onLayout={[Function]}
    style={
      [
        {
          "color": "#231B22",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Rubik",
          "fontSize": 16,
          "fontWeight": "500",
          "left": 0,
          "lineHeight": 24,
          "paddingRight": 20,
          "position": "absolute",
          "width": "",
        },
      ]
    }
    textWidth={null}
  >
    Title without Badge
  </Text>
  <View
    alignItems="center"
    bg="neon"
    borderRadius="circle"
    h="small"
    justifyContent="center"
    ml="xxxsmall"
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#DCFF79",
          "borderRadius": 9999,
          "height": 16,
          "justifyContent": "center",
          "marginLeft": 4,
          "width": 16,
        },
      ]
    }
    w="small"
  >
    <View
      aria-hidden={true}
      fill="#231B22"
      height={10.67}
      style={
        [
          {
            "height": 10.67,
            "width": 10.67,
          },
        ]
      }
      width={10.67}
    />
  </View>
</View>
`;
